multisig/src/lib.rs

```diff
@@ -59,32 +63,69 @@ pub enum MultiSigRequestAction {
     SetActiveRequestsLimit { active_requests_limit: u32 },
 }
 
-// The request the user makes specifying the receiving account and actions they want to execute (1 tx)
+/// The request the user makes specifying the receiving account and actions they want to execute (1 tx)
 #[derive(Clone, PartialEq, BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
 #[serde(crate = "near_sdk::serde")]
 pub struct MultiSigRequest {
     receiver_id: AccountId,
     actions: Vec<MultiSigRequestAction>,
 }
 
-// An internal request wrapped with the signer_pk and added timestamp to determine num_requests_pk and prevent against malicious key holder gas attacks
+/// An internal request wrapped with the signer_pk and added timestamp to determine num_requests_pk and prevent against malicious key holder gas attacks
 #[derive(Clone, PartialEq, BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
 #[serde(crate = "near_sdk::serde")]
 pub struct MultiSigRequestWithSigner {
     request: MultiSigRequest,
-    signer_pk: PublicKey,
+    member: String,
     added_timestamp: u64,
 }
 
+#[derive(Debug, BorshDeserialize, BorshSerialize, Clone, Eq, PartialEq)]
+pub enum MultisigMember {
+    AccessKey(PublicKey),
+    Account(AccountId),
+}
+
+impl From<MultisigMember> for String {
+    fn from(member: MultisigMember) -> Self {
+        match member {
+            MultisigMember::Account(account_id) => account_id,
+            MultisigMember::AccessKey(key) => String::from(&Base58PublicKey(key.to_vec())),
+        }
+    }
+}
+
+impl From<String> for MultisigMember {
+    fn from(s: String) -> Self {
+        if s.contains(":") {
```

Comment:
`:` is not a requirement for the public key string representation right now. This can lead to abuse if the base58 public key was only lowercase alphanumeric.

Reply:
I was thinking to add this as requirement.
But may be something like [{"account_id": "xyz"}, {"public_key": "zyx"}] is a better format just to make sure to avoid issues.